### PR TITLE
console, cli, cli-ext: query support for actions (close #4032)

### LINKED ISSUE #4032: support custom query fields through actions
Currently actions are only limited to mutations.

### PR DESCRIPTION
**Pull request by wawhal**
#### Console and cli-ext:
- Support query type actions

#### CLI 
- Support query type actions
- CLI only "appends" and does not "edit" the `actions.graphql` unless there's a metadata export.
- Remove GraphQL go dependency from CLI

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [ ] Server
- [x] Console
- [x] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System
- [ ] Tests
- [ ] Other (list it)

### Related Issues

#4032

### CODE REVIEW

File: console/src/shared/utils/sdlUtils.js [hunk 1]
```diff
@@ -146,26 +168,35 @@ const getActionFromMutationAstDef = astDef => {
...
     name: '',
     arguments: [],
     outputType: '',
     comment: '',
     error: null,
   };
+  let schemaAst;
+  try {
+    schemaAst = sdlParse(sdl);
+  } catch(_) {
```
Comment (beerose) [inline console/src/shared/utils/sdlUtils.js:181]: In a current version of the ECMAScript standard, you can do just:
```
catch {
}
```

File: console/src/shared/utils/deriveAction.js [hunk 2]
```diff
@@ -108,7 +111,7 @@ const deriveAction = (
   };
 
   const allHasuraTypes = clientSchema._typeMap;
-  const operationType = clientSchema._mutationType; // TODO better handling for queries
+  const operationType =  getOperationType(clientSchema, operation); // TODO better handling for queries
```
Comment (beerose) [inline console/src/shared/utils/deriveAction.js:114]: Is this comment still relevant?

File: cli-ext/src/services/sdl/to/handler.js [hunk 3]
```diff
@@ -42,7 +42,7 @@ const handlePayload = (payload) => {
   if (toDeriveOperation) {
     try {
       const derivation = deriveAction(toDeriveOperation, introspectionSchema, actionName);
-      const derivedActionSdl = getActionDefinitionSdl(derivation.action.name, derivation.action.arguments, derivation.action.output_type);
+      const derivedActionSdl = getActionDefinitionSdl(derivation.action.name, derivation.action.type, derivation.action.arguments, derivation.action.output_type);
```
Comment (beerose) [inline cli-ext/src/services/sdl/to/handler.js:45]: **[Question]** Shouldn't we also have prettier and eslint configured for `cli-ext/`?

Comment (wawhal) [reply cli-ext/src/services/sdl/to/handler.js:45]: Should. Soon.
https://github.com/hasura/graphql-engine/issues/4364

**Comment by beerose** [comment]
Random thought. If someone starts filling action definition and doesn't read what's below, and then while typing `Query` spots that something just disappeared, wouldn't it be confusing? 
What about disabling `kind`?
cc @rikinsk

**Comment by wawhal** [comment]
@beerose they'll just think that `kind` is not a thing for query type actions. That's usually the first thought. And it's correct. So I think it's fine.

**Comment by wawhal** [comment]
@arvi3411301 can you look at the CLI tests and also add it to CLI's PR review queue?

**Comment by shahidhk** [comment]
@wawhal Can you resolve the conflicts?

**Comment by fandy** [comment]
You guys are amazing.

**Pull request merged**